### DISCUSSION TITLE
How to syntax highlight LLVM tablegen .inc file

### DISCUSSION THREAD

**Discussion by hehaoqian** [clangd/clangd#1688] [Q&A] [answered]
Using VSCode clangd (0.1.24) with clangd 15.0.7 

The .inc file generated by LLVM tablegen is often not a complete header,
It need to be wrapped inside an actual header with `#define` to be actually used.

Taking `build/llvm-project/llvm/lib/Target/RISCV/RISCVGenSearchableTables.inc` as an example,
it contains

```
#ifdef GET_RISCVMaskedPseudosTable_DECL
......
#endif
```

which is used in `lib/Target/RISCV/RISCVISelDAGToDAG.h`

```
......
#define  GET_RISCVMaskedPseudosTable_DECL
#include "RISCVGenSearchableTables.inc"
```

If jumped into "RISCVGenSearchableTables.inc", 
everything is in grey without syntax highlight.

clangd is flawed that it process each file separately,
and .inc file is not a complete header, which needs to
be wrapped in another header, to have `#define`
and other necessary code before the inclusion of .inc file.

Any experience how to solve this?

**Comment by HighCommander4** [comment, marked as answer]
It would require fixing #45. You can read through the discussion there to get a sense of the challenges involved.

**Comment by hehaoqian** [comment]
It would be so nice if this is fixed.
Microsoft C/C++ extension and CLion supports this,
but they are laggy in large code base, such as LLVM.

And clangd is fundamentally broken in project that does not use complete header,
such as QEMU
